frontend/src/core/user/components/UserMenu.js

```diff
@@ -49,11 +50,29 @@ export class UserMenuBase extends React.Component<Props, State> {
     }
 
     render() {
-        const { parameters, signOut, user } = this.props;
+        const { parameters, signOut, user, isReadOnly } = this.props;
 
         const locale = parameters.locale;
         const project = parameters.project;
-        const tm_href = `/${locale}/${project}/${locale}.${project}.tmx`;
+        const resource = parameters.resource;
+
+        const tmHref = `/${locale}/${project}/${locale}.${project}.tmx`;
+        const transHref = `/download/?code=${locale}&slug=${project}&part=${resource}`;
+
+        const canDownload = (
+            project !== 'all-projects' &&
+            resource !== 'all-resources'
+        );
+        const canReview = (
+            user.translatorForLocales &&
+            user.translatorForLocales.includes(locale)
+        );
```

Comment:
This is taken from the [`History`](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/history/components/History.js#L60) component, but it's not really accurate: it doesn't take into account locales the user is a manager for (`managerForLocales`) and per-project translator permissions (`translatorForProjects`). See how [Translate.Current](https://github.com/mozilla/pontoon/blob/master/pontoon/base/static/js/translate.js#L4553) handles that.

We should amend `canReview` accordingly, and possibly store the information in `UserState` for easier reuse.

My question is, should we already on backend generate a list of (locale_code, project_slug) pairs and pass that to frontend? We'd do that using `managerForLocales`, `translatorForLocales` and `translatorForProjects`, which are all passed to frontend, but never used beyond the "`canReview`" use case.

Reply:
Added the [selector](https://github.com/mozilla/pontoon/pull/1226/commits/3c4cbd77d30ee3aac23e214effa1fe841ce958ba).